packages/angular-table/src/index.ts

```diff
@@ -1,96 +1,61 @@
-      const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(
-        () => {
-          return {
-            state: {},
-            onStateChange: () => {},
-            renderFallbackValue: null,
-            ...options(),
-          }
-        }
-      )
-
-      const notifier = signal([], { equal: () => false })
-      const table = createTable(untracked(resolvedOptionsSignal))
-      const state = signal(table.initialState)
-
-      function updateOptions() {
-        const tableState = untracked(state)
-        const resolvedOptions = untracked(resolvedOptionsSignal)
-        untracked(() => {
-          table.setOptions(prev => ({
-            ...prev,
-            ...resolvedOptions,
-            state: { ...tableState, ...resolvedOptions.state },
-            onStateChange: updater => {
-              const value =
-                updater instanceof Function ? updater(tableState) : updater
-              state.set(value)
-              resolvedOptions.onStateChange?.(updater)
-            },
-          }))
-        })
-      }
-
-      updateOptions()
+  options: () => TableOptions<TData>,
+  injector?: Injector
+): Table<TData> {
+  if (!injector) injector = inject(Injector)
+
+  return lazyInit(() => {
+    // Compose table resolved options as computed.
+    // This will allow the effect to be triggered when options are updated.
+    const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(() => ({
+      state: {},
+      onStateChange: () => {},
+      renderFallbackValue: null,
+      ...options(),
+    }))
+
+    const table = createTable(resolvedOptionsSignal())
+
+    // By default, manage table state here using the table's initial state
+    const state = signal<TableState>(table.initialState)
+
+    function updateOptions() {
+      const tableState = state()
+      const resolvedOptions = resolvedOptionsSignal()
+      table.setOptions(prev => ({
+        ...prev,
+        ...resolvedOptions,
+        state: { ...tableState, ...resolvedOptions.state },
+        onStateChange: updater => {
+          const value =
+            updater instanceof Function ? updater(tableState) : updater
+          state.set(value)
+          resolvedOptions.onStateChange?.(updater)
+        },
+      }))
+    }
 
-      let firstRender = true
-      effect(() => {
-        void [state(), resolvedOptionsSignal()]
-        if (firstRender) {
-          return (firstRender = false)
-        }
-        untracked(() => {
-          updateOptions()
-          notifier.set([])
-        })
-      })
+    // set table options again when options are updated
+    effect(() => updateOptions(), { injector })
 
-      const tableSignal = computed(
-        () => {
-          notifier()
-          return table
-        },
-        {
-          equal: () => false,
-        }
-      )
+    // set table options for the first time
+    updateOptions()
 
-      return proxifyTable(tableSignal)
```

Comment:
Removing proxifyTable doesn't allow in any way to listen to the table changes if you are using computed/effect. In a more realistic example, you may risk to use `ngDoCheck` or `getters` instead of `computed`

```ts
effect(() => {
  this.table() // on every table option/state change
})

effect(() => {
  this.table.getState() // called on every state change
})

effect(() => {
  this.table.getCenterRows() // called on every center rows change
})

Reply:
Got it. Working to put it back again. Thanks